pandas/core/frame.py

```diff
@@ -5272,10 +5272,15 @@ def combine(self, other, func, fill_value=None, overwrite=True):
         2  NaN  3.0 1.0
         """
         other_idxlen = len(other.index)  # save for compare
+        other_columns = other.columns
 
         this, other = self.align(other, copy=False)
         new_index = this.index
 
+        if other.empty and self.empty:
+            empty_columns = self.columns.union(other_columns)
+            return self.reindex(columns=empty_columns, index=[None]).dropna()
```

Comment:
why are you setting the index to [None] here, that is very weird. just reindex by the columns and should be good (and don't need the .dropna())

Reply:
okay will make the changes

Reply:
when I did reindex then second columns dtype changed to float64

Reply:
@jreback please have a look